pkg/controller/reconciler/deployment_reconciler.go

```diff
@@ -678,3 +691,28 @@ func (r *DeploymentReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.Secret{}).
 		Complete(r)
 }
+
+// OperationInProgress returns true if there is an operation in progress for the given deployment.
+func (r *DeploymentReconciler) OperationInProgress(annotations deploymentAnnotations) bool {
```

Comment:
Would it be clearer if these functions were defined on `deploymentAnnotations`? They don't need any state from the reconciler.